server/asset/btc/btc.go

```diff
@@ -254,6 +254,25 @@ func (btc *Backend) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 		}
 	}
 
+	txindex, err := btc.node.CheckTxIndex()
+	if err != nil {
+		if !isMethodNotFoundErr(err) {
+			btc.shutdown()
+			return nil, fmt.Errorf("%s getindexinfo check failed: %w", btc.name, err)
+		}
+		// Ignore and log err if getindexinfo method is not found.
+		// getindexinfo method is not currently supported by
+		// pre 0.21 versions of bitcoind, and some forks of
+		// bitcoin core (litecoin).
+		txindex = true
+		btc.log.Warnf("The getindexinfo RPC is unavailable for %s. Please ensure txindex is enabled in the node config.", btc.name)
+	}
+
+	if !txindex {
```

Comment:
Combine into an `else if` and remove the `txindex = true` in the `err != nil` branch above.

In fact, I would suggest limiting scope of the `txindex` variable like

```go
if txindex, err := btc.node.CheckTxIndex(); err != nil {
```